src/Core/Framework/Test/DependencyInjection/CompilerPass/EntityCompilerPassTest.php

```diff
@@ -39,4 +42,10 @@ public function testEntityRepositoryAutowiring(): void
         static::assertNotNull($container->getAlias('Shopware\Core\Framework\DataAbstractionLayer\EntityRepositoryInterface $customerRepository'));
         static::assertNotNull($container->getAlias('Shopware\Core\Framework\DataAbstractionLayer\EntityRepositoryInterface $customerAddressRepository'));
     }
+
+    public function testEntityRepositoryAutowiringForAlreadyDefinedRepositories(): void
+    {
+        static::assertTrue($this->getContainer()->has('product.repository'));
```

Comment:
Please create a new container builder with the services and process that :) like above as this test should unbound from actual build container

Reply:
Okay.

I made the assumption that since `$productRepository = $this->getContainer()->get('product.repository')` is used everywhere in tests, that will be enough.